### PR TITLE
Feature/initial-pipe

### PR DESCRIPTION
**Pull request by AdamJHall**
This PR creates the initial configuration and docker container for a phpstan pipe for bitbucket pipelines. This mainly uses code from the `aligent/code-standards-pipe-php` to only execute on files changed between the current and target branch.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+ARG PHP_VERSION
+FROM php:${PHP_VERSION}-cli-alpine
+
+# Fetch the latest version of composer
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
```
Comment (hubertott) [inline Dockerfile:5]: Cool! I need to remember we can do this 😮

File: README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+```yaml
+      - step:
+          name: "PHPStan check"
+          script:
+            - pipe: aligent/phpstan-pipe:latest
+              variables:
+                SKIP_DEPENDENCIES: "false"
+                MAGENTO_USER: "USER"
+                MAGENTO_PASS: "PASS"
```
Comment (hubertott) [inline README.md]: Should we be suggesting adding Magento Repo credentials as Repository Secrets instead?

File: pipe/pipe.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+     if $DEBUG; then
+          echo "State of working directory"
+          git status
+     fi
+
+     echo "Comparing HEAD against branch $TARGET_BRANCH"
+     MERGE_BASE=$(git merge-base HEAD $TARGET_BRANCH)
+
+     CHANGED_FILES=$(git diff --relative --name-only --diff-filter=AM $MERGE_BASE -- '*.php' '*.phtml')
```
Comment (hubertott) [inline pipe/pipe.sh]: This process seems to assume PHPStan has been run at least once on the whole repo.

Should that be documented or add a flag to force a full scan?

**Review by hubertott — COMMENTED** [review]
This looks good. Just a few things to think about from me.

File: pipe/pipe.sh (hunk 3 above)
Comment (AdamJHall) [reply pipe/pipe.sh]: A flag for a full scan is probably a good idea, the idea behind the changed files though is that we don't want the devs to have to go and make a HUGE amount of changes when introducing phpstan. 

Instead if we only do it on new PRs they will slowly get used to it and touch more and more code, then we hit them with the big scan :D

**Comment by AdamJHall** [comment]
@hubertott,

Added a SCAN_DIRECTORY variable based on your feedback :)

**Review by hubertott — APPROVED** [review]
Nice!

**Pull request merged**